docker build fails on ARM64

The Dockerfile fails to build on ARM64 machines (a Macbook Pro with Apple M1 Pro processor). 

Building an amd64 image works fine so not a huge issue really. I added a PR that adds the `platform: linux/amd64` to `docker-compose.yml` to make the docker instructions work for people with new Apple machines here: https://github.com/gravitystorm/openstreetmap-carto/pull/4539.

### Expected behavior
`docker compose build kosmtik` finishes successfully on an ARM64 machine. 

Building an x86/amd64 image works: `docker build . --platform linux/amd64  -t openstreetmap-carto:dev` finishes without  error. 

### Actual behavior

`docker build .`, and therefore also `docker compose build`, crashes with the output below.

It appears there are no prebuilt binaries for mapnik/v3.7.2/node-v57 for arm64 from mapbox, it attempts compiling and it's downhill from there.

### Links and screenshots illustrating the problem

```
frehoy@fhmbp openstreetmap-carto % docker build . --progress plain --tag openstreetmap-carto:dev
#1 [internal] load build definition from Dockerfile
#1 sha256:22a728aa5c43c56d86db50b7e80136c92d6dc0d0edcd886f5fc2b7efbb5849b0
#1 transferring dockerfile: 37B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:41a2fe275de6ca9b7d952a5a67e5a2b91217438b6d3ba51640ada93dfb1b2efc
#2 transferring context: 34B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/ubuntu:bionic
#3 sha256:9481ca12cc018c3244ae38fbc1bc7a5a03be470741643979e92f7369e6b2ef52
#3 DONE 2.0s

#4 [1/7] FROM docker.io/library/ubuntu:bionic@sha256:627b1184c9100a22ba9dcf531908b9e24af99fa54e45c10f57852fb890a57ea6
#4 sha256:d76a875fd2bd87ee3547ea913c663965841ecd17783c787b100db9b41e607f52
#4 DONE 0.0s

#5 [2/7] RUN apt-get update && apt-get install --no-install-recommends -y     ca-certificates curl gnupg postgresql-client python3 python3-distutils     fonts-hanazono fonts-noto-cjk fonts-noto-hinted fonts-noto-unhinted     mapnik-utils nodejs npm ttf-unifont unzip && rm -rf /var/lib/apt/lists/*
#5 sha256:fc46f0f92fce8be983f34238cd62a45d675d661cd52fe2677240cd7adf4b1872
#5 CACHED

#6 [3/7] RUN npm set prefix /usr && npm install -g kosmtik
#6 sha256:c86556152daeaacdc3e87f4c37243ee6f8c23aa6cb49413669ef01294973f1dc
#6 2.354 npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
#6 3.079 npm WARN deprecated nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
#6 24.72 npm WARN deprecated protozero@1.5.1: protozero should no longer be used via npm, install instead via https://github.com/mapbox/mason
#6 27.49 npm WARN deprecated har-validator@5.1.5: this library is no longer supported
#6 27.64 npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
#6 33.21 /usr/bin/kosmtik -> /usr/lib/node_modules/kosmtik/index.js
#6 33.67
#6 33.67 > mapnik@3.7.2 install /usr/lib/node_modules/kosmtik/node_modules/mapnik
#6 33.67 > node-pre-gyp install --fallback-to-build
#6 33.67
#6 34.46 node-pre-gyp ERR! Tried to download(403): https://mapbox-node-binary.s3.amazonaws.com/mapnik/v3.7.2/node-v57-linux-arm64-Release.tar.gz
#6 34.46 node-pre-gyp ERR! Pre-built binaries not found for mapnik@3.7.2 and node@8.10.0 (node-v57 ABI, glibc) (falling back to source compile with node-gyp)
#6 35.07 gyp WARN EACCES user "undefined" does not have permission to access the dev dir "/root/.node-gyp/8.10.0"
#6 35.07 gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/kosmtik/node_modules/mapnik/.node-gyp"
#6 35.42 /bin/sh: 1: mapnik-config: not found
#6 35.42 gyp: Call to 'mapnik-config --cflags' returned exit status 127 while in binding.gyp. while trying to load binding.gyp
#6 35.42 gyp ERR! configure error
#6 35.42 gyp ERR! stack Error: `gyp` failed with exit code: 1
#6 35.42 gyp ERR! stack     at ChildProcess.onCpExit (/usr/lib/node_modules/kosmtik/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
#6 35.42 gyp ERR! stack     at emitTwo (events.js:126:13)
#6 35.42 gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
#6 35.42 gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
#6 35.42 gyp ERR! System Linux 5.10.104-linuxkit
#6 35.42 gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/kosmtik/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--module=/usr/lib/node_modules/kosmtik/node_modules/mapnik/lib/binding/mapnik.node" "--module_name=mapnik" "--module_path=/usr/lib/node_modules/kosmtik/node_modules/mapnik/lib/binding"
#6 35.42 gyp ERR! cwd /usr/lib/node_modules/kosmtik/node_modules/mapnik
#6 35.42 gyp ERR! node -v v8.10.0
#6 35.42 gyp ERR! node-gyp -v v3.6.0
#6 35.42 gyp ERR! not ok
#6 35.43 node-pre-gyp ERR! build error
#6 35.43 node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /usr/lib/node_modules/kosmtik/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/usr/lib/node_modules/kosmtik/node_modules/mapnik/lib/binding/mapnik.node --module_name=mapnik --module_path=/usr/lib/node_modules/kosmtik/node_modules/mapnik/lib/binding' (1)
#6 35.43 node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/kosmtik/node_modules/mapnik/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
#6 35.43 node-pre-gyp ERR! stack     at emitTwo (events.js:126:13)
#6 35.43 node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
#6 35.43 node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:925:16)
#6 35.43 node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
#6 35.43 node-pre-gyp ERR! System Linux 5.10.104-linuxkit
#6 35.43 node-pre-gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/kosmtik/node_modules/mapnik/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
#6 35.43 node-pre-gyp ERR! cwd /usr/lib/node_modules/kosmtik/node_modules/mapnik
#6 35.43 node-pre-gyp ERR! node -v v8.10.0
#6 35.43 node-pre-gyp ERR! node-pre-gyp -v v0.6.39
#6 35.43 node-pre-gyp ERR! not ok
#6 35.43 Failed to execute '/usr/bin/node /usr/lib/node_modules/kosmtik/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/usr/lib/node_modules/kosmtik/node_modules/mapnik/lib/binding/mapnik.node --module_name=mapnik --module_path=/usr/lib/node_modules/kosmtik/node_modules/mapnik/lib/binding' (1)
#6 38.26 /usr/lib
#6 38.26 `-- (empty)
#6 38.26
#6 38.26 npm ERR! Linux 5.10.104-linuxkit
#6 38.26 npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install" "-g" "kosmtik"
#6 38.26 npm ERR! node v8.10.0
#6 38.26 npm ERR! npm  v3.5.2
#6 38.26 npm ERR! code ELIFECYCLE
#6 38.26
#6 38.26 npm ERR! mapnik@3.7.2 install: `node-pre-gyp install --fallback-to-build`
#6 38.26 npm ERR! Exit status 1
#6 38.26 npm ERR!
#6 38.26 npm ERR! Failed at the mapnik@3.7.2 install script 'node-pre-gyp install --fallback-to-build'.
#6 38.26 npm ERR! Make sure you have the latest version of node.js and npm installed.
#6 38.26 npm ERR! If you do, this is most likely a problem with the mapnik package,
#6 38.26 npm ERR! not with npm itself.
#6 38.26 npm ERR! Tell the author that this fails on your system:
#6 38.26 npm ERR!     node-pre-gyp install --fallback-to-build
#6 38.26 npm ERR! You can get information on how to open an issue for this project with:
#6 38.26 npm ERR!     npm bugs mapnik
#6 38.26 npm ERR! Or if that isn't available, you can get their info via:
#6 38.26 npm ERR!     npm owner ls mapnik
#6 38.26 npm ERR! There is likely additional logging output above.
#6 38.27
#6 38.27 npm ERR! Please include the following file with any support request:
#6 38.27 npm ERR!     /npm-debug.log
#6 38.27 npm ERR! code 1
#6 ERROR: executor failed running [/bin/sh -c npm set prefix /usr && npm install -g kosmtik]: exit code: 1
------
 > [3/7] RUN npm set prefix /usr && npm install -g kosmtik:
------
executor failed running [/bin/sh -c npm set prefix /usr && npm install -g kosmtik]: exit code: 1
```

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -2,6 +2,7 @@ version: '2'
 services:
   kosmtik:
     image: kosmtik:v1
+    platform: linux/amd64
     build:
       context: .
       dockerfile: Dockerfile

